src/app/dynamic/enterprise/quotas/component.spec.ts

```diff
@@ -0,0 +1,267 @@
+
+    expect(element.nativeElement.hidden).toEqual(true);
+  });
+
+  it('should not display table when quotas are loaded empty', async () => {
+    component.isLoading = false;
+    component.quotas = [];
+    fixture.detectChanges();
+
+    const element = fixture.debugElement.query(By.css('table'));
+
+    expect(element.nativeElement.hidden).toEqual(true);
+  });
+
+  it('should not display loading spinner when quotas are loading', () => {
+    component.isLoading = true;
+    fixture.detectChanges();
+
+    const element = fixture.debugElement.query(By.css('#quotas-loading-spinner'));
+
+    expect(element).toBeTruthy();
+  });
+
+  it('should hide loading spinner when quotas are loaded', () => {
+    component.isLoading = false;
+    fixture.detectChanges();
+
+    const element = fixture.debugElement.query(By.css('#quotas-loading-spinner'));
+
+    expect(element).toBeFalsy();
+  });
+
+  it('should display correct message when loaded quotes are empty', () => {
+    component.isLoading = false;
+    component.quotas = [];
+    fixture.detectChanges();
+
+    const element = fixture.debugElement.query(By.css('#quotes-not-found'));
+
+    expect(element.nativeElement.textContent.trim()).toEqual('No Quotas Found');
+  });
+
+  it('should display paginator when received quotes are more than the user defined page size', () => {
+    const mockLength = 5;
+    component.quotas.length = mockLength - 1;
+    component.paginator.pageSize = mockLength;
+    fixture.detectChanges();
+
+    const element = fixture.debugElement.query(By.css('#quotas-paginator'));
+
+    expect(element.nativeElement.hidden).toEqual(true);
+  });
+
+  it('should hide paginator when received quotes are less than the user defined page size', () => {
+    const mockLength = 5;
+    component.quotas.length = mockLength;
+    component.paginator.pageSize = mockLength + 1;
+    fixture.detectChanges();
+
+    const element = fixture.debugElement.query(By.css('#quotas-paginator'));
+
+    expect(element.nativeElement.hidden).toEqual(true);
+  });
+
+  it('should set quotas when ngOnInit is called', async () => {
+    const quotas = await getQuotas();
+
+    component.ngOnInit();
+
+    expect(component.quotas).toEqual(quotas);
+  });
+
+  it('should set user when ngOnInit is called', async () => {
+    component.user = null;
+    const user = await lastValueFrom(userService.currentUser);
+
+    component.ngOnInit();
+
+    expect(component.user).toStrictEqual(user);
+  });
+
+  it('should set paginator.pageSize when ngOnInit is called', async () => {
+    const userSettings = await lastValueFrom(userService.currentUserSettings);
+
+    component.ngOnInit();
+
+    expect(component.paginator.pageSize).toEqual(userSettings.itemsPerPage);
+  });
+
+  it('should render table rows when quotas are set in dataSource.data is set', async () => {
+    await setQuotas();
+
+    const elements = fixture.debugElement.nativeElement.querySelectorAll('.mat-row');
```

Comment:
Alwas go from Specific -> General  

there could be many `.mat-rows` within page it doesn't mean only page you are showing will have `.mat-row` what if we have table in DOM which isn hidden

https://github.com/kubermatic/dashboard/pull/4641/files#r920254258

You can use table ID -> find ROWS within **EXPLICIT** table

thoughts?

Update all places IMO, if you have any reason let me know

Reply:
These components are lazy loaded, which means there won't be any other table, unless we add another one.